ts/test/session/unit/swarm_polling/SwarmPolling_pollForAllKeys_test.ts

```diff
@@ -0,0 +1,570 @@
+    stubData('saveConversation').resolves();
+    stubData('getSwarmNodesForPubkey').resolves();
+    stubData('getLastHashBySnode').resolves();
+
+    Sinon.stub(Convo, 'commitConversationAndRefreshWrapper').resolves();
+
+    TestUtils.stubLibSessionWorker(undefined);
+
+    Sinon.stub(SnodePool, 'getSwarmFor').resolves(generateFakeSnodes(5));
+    Sinon.stub(SnodeAPIRetrieve, 'retrieveNextMessagesNoRetries').resolves([]);
+
+    TestUtils.stubWindow('inboxStore', undefined);
+    TestUtils.stubWindow('getGlobalOnlineStatus', () => true);
+    TestUtils.stubWindowLog();
+
+    const convoController = ConvoHub.use();
+    await convoController.load();
+    ConvoHub.use().getOrCreate(ourPubkey.key, ConversationTypeEnum.PRIVATE);
+
+    swarmPolling = getSwarmPollingInstance();
+    swarmPolling.resetSwarmPolling();
+    pollOnceForKeySpy = Sinon.spy(swarmPolling, 'pollOnceForKey');
+
+    clock = sinon.useFakeTimers({ now: Date.now(), shouldAdvanceTime: true });
+    stubData('createOrUpdateItem').resolves();
+  });
+
+  afterEach(() => {
+    Sinon.restore();
+    ConvoHub.use().reset();
+    clock.restore();
+    resetHardForkCachedValues();
+  });
+
+  it('does run for our pubkey even if activeAt is really old ', async () => {
+    stubWithGroups([]);
+    stubWithLegacyGroups([]);
+    const convo = ConvoHub.use().getOrCreate(ourNumber, ConversationTypeEnum.PRIVATE);
+    convo.set('active_at', Date.now() - 1000 * 3600 * 25);
+    await swarmPolling.start(true);
+
+    expect(pollOnceForKeySpy.callCount).to.eq(1);
+    expect(pollOnceForKeySpy.firstCall.args).to.deep.eq(pollOnceForUsArgs(ourPubkey.key));
+  });
+
+  it('does run for our pubkey even if activeAt is recent ', async () => {
+    stubWithGroups([]);
+    stubWithLegacyGroups([]);
+    const convo = ConvoHub.use().getOrCreate(ourNumber, ConversationTypeEnum.PRIVATE);
+    convo.set('active_at', Date.now());
+    await swarmPolling.start(true);
+
+    expect(pollOnceForKeySpy.callCount).to.eq(1);
+    expect(pollOnceForKeySpy.firstCall.args).to.deep.eq(pollOnceForUsArgs(ourPubkey.key));
+  });
+
+  describe('legacy group', () => {
+    it('does run for group pubkey on start no matter the recent timestamp', async () => {
+      const groupPk = TestUtils.generateFakePubKeyStr();
+      const convo = ConvoHub.use().getOrCreate(groupPk, ConversationTypeEnum.GROUP);
+      stubWithLegacyGroups([groupPk]);
+      stubWithGroups([]);
+      convo.set('active_at', Date.now());
+      const groupConvoPubkey = PubKey.cast(groupPk);
+      swarmPolling.addGroupId(groupConvoPubkey);
+      await swarmPolling.start(true);
+
+      // our pubkey will be polled for, hence the 2
+      expect(pollOnceForKeySpy.callCount).to.eq(2);
+
+      expect(pollOnceForKeySpy.firstCall.args).to.deep.eq(pollOnceForUsArgs(ourPubkey.key));
+      expect(pollOnceForKeySpy.secondCall.args).to.deep.eq(pollOnceForGroupLegacyArgs(groupPk));
+    });
+
+    it('does only poll from -10 for closed groups', async () => {
```

Comment:
How do you know this is only -10? Sorry I can't tell from reading the test

Reply:
legacy groups only ever needed to poll from -10, as that was the only namespace allowing unauthenficated retrieval of messages.